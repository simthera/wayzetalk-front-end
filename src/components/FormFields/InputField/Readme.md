# enabled input example

<!-- ```js
<InputField name="test" autoComplete="off" type="text" disabled={false} placeholder="test field" />
```

# disabled input example

```js
<InputField name="test" autoComplete="off" type="text" disabled={false} placeholder="test field" disabled/>
``` -->
