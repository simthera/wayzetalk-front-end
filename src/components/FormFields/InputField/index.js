/**
 *
 * InputField
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import FormFieldError from '../FormFieldError';
import {
  StyledInputField,
  StyledLabel,
  StyledDropdownListWrapper,
  StyledDropdownListWrapperInner,
} from './styledComponents/styledInputField';

function InputField(props) {
  const {
    input,
    type,
    placeholder,
    value,
    disabled = false,
    multiline = false,
    autoComplete = 'on',
    rows = 1,
    rowsMax = 1,
    meta,
    onFocus,
    onBlur,
    isDropDownVisible = false,
    options,
    name,
    onChange,
    handleSettingOfFields,
    hasDropDownOptions = false,
    fieldType = 'input',
  } = props;

  const handleFocus = () => {
    if (onFocus && typeof onFocus === 'function') {
      onFocus();
    }
  };

  const handleBlur = () => {
    if (onBlur && typeof onBlur === 'function') {
      onBlur();
    }
  };
  const handleChange = event => {
    if (onChange && typeof onChange === 'function') {
      onChange(event.target.value);
    }
  };

  return (
    <>
      <StyledLabel
        htmlFor={input && input.name}
        hasError={meta && meta.touched && meta.error}
      >
        <StyledInputField
          {...input}
          as={fieldType}
          type={type}
          placeholder={placeholder}
          label={placeholder}
          rows={rows}
          value={value}
          onFocus={handleFocus}
          onBlur={handleBlur}
          onChange={handleChange}
          rowsMax={rowsMax}
          multiline={multiline}
          autoComplete={autoComplete}
          disabled={disabled}
          hasError={meta && meta.touched && meta.error}
        />

        {isDropDownVisible && hasDropDownOptions && options.length > 0 && (
          <StyledDropdownListWrapper>
            <StyledDropdownListWrapperInner>
              {options.map((option, idx) => (
                <a
                  href="/"
                  key={idx}
                  onClick={e => {
                    e.preventDefault();
                    handleSettingOfFields(option);
                  }}
                >
                  {option.formattedAddress}
                </a>
              ))}
            </StyledDropdownListWrapperInner>
          </StyledDropdownListWrapper>
        )}
      </StyledLabel>
      {meta && meta.touched && meta.error && (
        <FormFieldError error={meta.error} />
      )}
    </>
  );
}

InputField.propTypes = {
  input: PropTypes.object,
  type: PropTypes.string,
  value: PropTypes.string,
  placeholder: PropTypes.string,
  autoComplete: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  disabled: PropTypes.bool,
  meta: PropTypes.object,
  rows: PropTypes.string,
  rowsMax: PropTypes.string,
  onFocus: PropTypes.func,
  onBlur: PropTypes.func,
  onChange: PropTypes.func,
  handleSettingOfFields: PropTypes.func,
};

export default InputField;
