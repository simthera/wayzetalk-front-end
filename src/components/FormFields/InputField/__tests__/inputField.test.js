import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import InputField from '../index';

afterEach(cleanup);

test('does component render', () => {
  const { asFragment } = render(
    <InputField name="test" type="text" onChange={() => alert('changed')} />, // eslint-disable-line
  );
  expect(asFragment()).toMatchSnapshot();
});
