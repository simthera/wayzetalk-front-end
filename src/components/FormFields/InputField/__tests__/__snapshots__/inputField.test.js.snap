// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does component render 1`] = `
<DocumentFragment>
  <div>
    <label
      for="test"
    >
      <input
        autocomplete="on"
        class="sc-bwzfXH uTxCW"
        rows="1"
      />
    </label>
  </div>
</DocumentFragment>
`;
